etcdmain/grpc_proxy.go

```diff
@@ -211,6 +223,12 @@ func startGRPCProxy(cmd *cobra.Command, args []string) {
 	go func() { errc <- srvhttp.Serve(httpl) }()
 	go func() { errc <- m.Serve() }()
 	if len(grpcProxyMetricsListenAddr) > 0 {
+		if grpcProxyMetricsListenCert != "" && grpcProxyMetricsListenKey != "" {
+			tlsinfo = newTLS(grpcProxyMetricsListenCA, grpcProxyMetricsListenCert, grpcProxyMetricsListenKey)
```

Comment:
@ironcladlou said `For the health listener, honor global the global TLS configuration by default (for compatibility with today's behavior), but provide a new flag to allow control over health endpoint client certificate auth (e.g. --health-client-cert-auth=false).` in issue #11993.  
i agrees with it. can we add a flag to access /health,/proxy/health without cert auth?

Reply:
> pr #12114 adds health handler for grpcproxy self, if grpcProxyMetricsListenAddr uses own certificate for metrics endpoint, we should also update healthcheck client certs, otherwise, it is failed to access '/proxy/health' handle.

I already mentioned this [before](https://github.com/etcd-io/etcd/pull/10504/files#r261113302). Therefore, when using separate certificates for metrics, a warning is displayed.

> i agrees with it. can we add a flag to access /health,/proxy/health without cert auth?

Do you want the server to serve some endpoints without certificates? Theoretically, this is possible, but will require rewriting the server. To do this, you will need to make authorization optional and do it not at the TLS level. This is out of scope of this proposal.

Reply:
ok. thanks. I will try to add --health-client-cert-auth flag in another PR. thanks.

Reply:
@tangcong Can this PR be merged?

Reply:
lgtm. can you add changelog and doc in another pr? thanks. @hexfusion can you take a final look? thanks.

Reply:
@tangcong sorry for delay yes I will take a look.